### PR TITLE
#165044465 Setup API

### PR DESCRIPTION
**Pull request by oleesir**
#### What does this PR do?
 Implement setup-api
#### Description of Task to be completed?
 Implement setup-api
#### What are the relevant pivotal tracker stories?
[#165044465](https://www.pivotaltracker.com/story/show/165044465)
#### Any background context you want to add?
N/A
#### Screenshots
N/A

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "presets": ["@babel/preset-env"]
+}
```
Comment (iverenshaguy) [inline .babelrc]: Please always add a newline at the end of every file

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      "error",
+      {
+        "require": {
+          "FunctionDeclaration": true,
+          "MethodDefinition": true,
+          "ClassDeclaration": true
+        }
+      }
+    ]
+
```
Comment (iverenshaguy) [inline .eslintrc]: Please remove this space

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+}
```
Comment (iverenshaguy) [inline .eslintrc]: New line

File: package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "banka",
+  "version": "1.0.0",
+  "description": "A banking application that helps users perform banking transactions.",
+  "main": "index.js",
+  "scripts": {
+    "build": "babel server/src -d server/dist",
+    "dev": "nodemon ./server/src/app.js --exec babel-node",
+    "start": "nodemon --exec babel-node src/app.js",
```
Comment (iverenshaguy) [inline package.json]: Why are you still starting the app with `nodemon` and `babel-node`? You should be starting the built app instead by using just `node`. Kindly update this

File: package.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  "devDependencies": {
+    "@babel/cli": "^7.4.4",
+    "@babel/core": "^7.4.4",
+    "@babel/node": "^7.2.2",
+    "@babel/preset-env": "^7.4.4",
+    "@babel/register": "^7.4.4",
+    "chai": "^4.2.0",
+    "eslint": "^5.3.0",
+    "eslint-config-airbnb-base": "^13.1.0",
+    "eslint-plugin-import": "^2.17.2",
```
Comment (iverenshaguy) [inline package.json:32]: This was not used in the `eslint` config. What was your reason for adding the package and what do you want to use it for?

**Review by iverenshaguy — CHANGES_REQUESTED** [review]

**Comment by iverenshaguy** [comment]
Good job. Are you supposed to set up your continuous development apps/badge now i.e CodeClimate, Coveralls/CodeCov (I prefer CodeCov since it covers everything and does not need too much manipulation), TravisCI etc?

File: package.json (hunk 4 above)
Comment (oleesir) [reply package.json]: noted

File: package.json (hunk 5 above)
Comment (oleesir) [reply package.json:32]: they said we should use the airbnb style guide, so i installed this[ npx install-peerdeps --dev eslint-config-airbnb-base] to give me the latest dependencies

File: .eslintrc (hunk 3 above)
Comment (oleesir) [reply .eslintrc]: noted

File: .babelrc (hunk 1 above)
Comment (oleesir) [reply .babelrc]: noted

File: .eslintrc (hunk 2 above)
Comment (oleesir) [reply .eslintrc]: noted

File: package.json (hunk 4 above)
Comment (oleesir) [reply package.json]: do you mean this  "start": "node ./server/src/app.js"

Comment (iverenshaguy) [reply package.json]: Can you start directly from the source folder using node?

File: package.json (hunk 5 above)
Comment (iverenshaguy) [reply package.json:32]: Great!

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1 @@
+node_modules
```
Comment (iverenshaguy) [inline .eslintignore]: add a new line

**Comment by oleesir** [comment]
> Good job. Are you supposed to set up your continuous development apps/badge now i.e CodeClimate, Coveralls/CodeCov (I prefer CodeCov since it covers everything and does not need too much manipulation), TravisCI etc?

> I will do it in the next branch

**Review by iverenshaguy — APPROVED** [review]

**Pull request merged**